model/project_parser_merge_functions.go

```diff
@@ -188,27 +188,27 @@ func (pp *ParserProject) mergeUnique(toMerge *ParserProject) error {
 func (pp *ParserProject) mergeBuildVariant(toMerge *ParserProject) error {
 	catcher := grip.NewBasicCatcher()
 
-	bvs := map[string]*parserBV{}
+	bvs := map[string]parserBV{}
 	for _, bv := range pp.BuildVariants {
 		newBv := bv
-		bvs[bv.Name] = &newBv
+		bvs[bv.Name] = newBv
 	}
 	for _, bv := range toMerge.BuildVariants {
-		if _, ok := bvs[bv.Name]; ok {
+		if currentBV, ok := bvs[bv.Name]; ok {
 			if !bv.canMerge() {
 				catcher.Errorf("build variant '%s' has been declared already", bv.Name)
 			} else {
-				bvs[bv.Name].Tasks = append(bvs[bv.Name].Tasks, bv.Tasks...)
-				bvs[bv.Name].DisplayTasks = append(bvs[bv.Name].DisplayTasks, bv.DisplayTasks...)
+				currentBV.Tasks = append(bvs[bv.Name].Tasks, bv.Tasks...)
+				currentBV.DisplayTasks = append(bvs[bv.Name].DisplayTasks, bv.DisplayTasks...)
+				bvs[bv.Name] = currentBV
 			}
 		} else {
-			pp.BuildVariants = append(pp.BuildVariants, bv)
```

Comment:
This line is confusing me a bit. Why was it here before and why do we no longer want it?

Reply:
I think it's just an oversight; we set pp.BuildVariants to empty at 209 and we don't use it at all within the loop so there's no reason to do this.

Reply:
its because i wrote it when i was young and foolish 😭